File: Assets/Scripts/MissileController.cs
using UnityEngine;
using System.Collections;

public class MissileController : MonoBehaviour 
{
    public float Speed = 3F;
    public string PlayerTag = "Player";
	public float ExplosionScale = 2F;
	private Animator _anim;
	private bool _hasHit = false;

	private void Start()
	{
		_anim = GetComponent<Animator>();
	}

	private void Update () 
	{
		if (!_hasHit)
		{
			transform.position += transform.up * Speed * Time.deltaTime;
		}
	}

	private void OnTriggerEnter2D(Collider2D other)
    {
		if(other.tag != PlayerTag && other.tag != "Border")
		{
			StartCoroutine(StartExplosion());
		}
    }

	private void OnTriggerExit2D(Collider2D other)
    {
        if (other.tag == "Border")
        {
            Destroy(this.gameObject);
        }
    }

	private IEnumerator StartExplosion()
	{
		_hasHit = true;
		_anim.Play("Explosion1");
		this.GetComponent<BoxCollider2D>().enabled = false;
		this.transform.localScale = Vector3.one * ExplosionScale;
		yield return new WaitForSeconds(1f);
		Destroy(this.gameObject);
	}
}

File: Assets/Scripts/JS/MissileController.js
#pragma strict

public var speed = 2F;
public var playerTag = "Player";

function Update () 
{
    transform.position += transform.up * speed * Time.deltaTime;
}

function OnTriggerEnter2D(other: Collider2D)
{
    if(other.tag != playerTag && other.tag != "Border")
    {
        Destroy(this.gameObject);
    } 
    else if (other.tag == "Player" || other.tag == "Player2")
    {
    	print("hit");
    }
}

function OnTriggerExit2D(other: Collider2D)
{
    if (other.tag == "Border")
    {
        Destroy(this.gameObject);
    }
}